### PR TITLE
golang mod fix

### PR DESCRIPTION
**Pull request by irisdingbj**
## Fixes Issue #

-  Json parsing issues where I have `{ ` inside  the `go list -m -json`  result, eg: 

```
{         "Path": "github.com/Microsoft/hcsshim",
        "Version": "v0.8.7",
        "Replace": {
                "Path": "github.com/Microsoft/hcsshim",
                "Version": "v0.8.8-0.20200421182805-c3e488f0d815",
                "Time": "2020-04-21T18:28:05Z",
                "GoMod": "/xxxxxx/go/pkg/mod/cache/download/github.com/!microsoft/hcsshim/@v/v0.8.8-0.20200421182805-c3e488f0d815.mod"
        },         "Indirect": true,
        "GoMod": "/xxxxxx/pkg/mod/cache/download/github.com/!microsoft/hcsshim/@v/v0.8.8-0.20200421182805-c3e488f0d815.mod" }

``` 

- Add `declared-license` into the dependencies.

- Correct the `FilePath` , 'FileName`,'ActualFilePath` for the found go modules from the `Dir`

- Respect `analyzer.golang.path` in `dependencycheck.properties`  as 1st priority and backward to default if we have no such setting. 

.

### CODE REVIEW

File: core/src/main/java/org/owasp/dependencycheck/data/golang/GoModDependency.java [hunk 1]
```diff
@@ -155,9 +174,33 @@ private Dependency createDependency(Dependency parentDependency, String name, St
...
     }
 
+	private void extractLicense(Dependency dependency, File file) {
+
+		File[] files = file.listFiles();
+		if (files != null ) {
+			for (File f : files) {
+				if (supportedLicenseFilenames.contains(f.getName())) {
+					String license = f.getAbsolutePath();
+					dependency.setLicense(license);
```
Comment (jeremylong) [inline core/src/main/java/org/owasp/dependencycheck/data/golang/GoModDependency.java]: Did you mean the license to be just the path to the file? Or should we read the content of the file into the license text in the dependency object?

File: core/src/main/java/org/owasp/dependencycheck/analyzer/GolangModAnalyzer.java [hunk 2]
```diff
@@ -132,22 +133,18 @@ protected FileFilter getFileFilter() {
...
-                        );
-                        goPath = "go";
-                    }
-                }
+                    LOGGER.warn("Provided path to `go` executable is invalid. Trying default location. "
+                            + "If you do want to set it, please set the `{}` property",
+                            Settings.KEYS.ANALYZER_GOLANG_PATH
+                    );
+                    goPath = "go";
+               }
```
Comment (jeremylong) [inline core/src/main/java/org/owasp/dependencycheck/analyzer/GolangModAnalyzer.java]: I don't believe this change is needed as this is contained in a synchronized block. We only need to set the goPath once.

**Review by jeremylong — COMMENTED** [review]
Thanks for the PR - some good things in here. However, I had a PR in flight that changed some of what you were modifying. I already have a change in flight to incorporate some of you suggestions in the PR; as such there is no need to rebase and update this PR. However, I do have a question around your intent with the license file - see comment/question in the code...

Regarding the fix for the `Replace {}` issue, which does correctly have a trailing comma, I have modified the JsonArrayFixingInputStream to correctly handle a closing brace with a trailing comma (I'll push the change shortly).

File: core/src/main/java/org/owasp/dependencycheck/analyzer/GolangModAnalyzer.java (hunk 2 above)
Comment (irisdingbj) [reply core/src/main/java/org/owasp/dependencycheck/analyzer/GolangModAnalyzer.java]: we have `private static String goPath = "go"; `  Does this mean the setting from `dependencycheck.properties` will not take effect at all?

File: core/src/main/java/org/owasp/dependencycheck/data/golang/GoModDependency.java (hunk 1 above)
Comment (irisdingbj) [reply core/src/main/java/org/owasp/dependencycheck/data/golang/GoModDependency.java]: Yes, I am hesitant for this part as well..  Now I just set   it as filepath to the license file.    How do we deal with this for other analyzer? What is the suggested approach for this?

**Comment by irisdingbj** [comment]
> Thanks for the PR - some good things in here. However, I had a PR in flight that changed some of what you were modifying. I already have a change in flight to incorporate some of you suggestions in the PR; as such there is no need to rebase and update this PR. However, I do have a question around your intent with the license file - see comment/question in the code...
> 
> Regarding the fix for the `Replace {}` issue, which does correctly have a trailing comma, I have modified the JsonArrayFixingInputStream to correctly handle a closing brace with a trailing comma (I'll push the change shortly).

Thanks @jeremylong  for the review! Can you please put a link for the fix you are working on? Thanks!

**Comment by irisdingbj** [comment]
@jeremylong  Can we get this PR in? Thanks!

**Pull request merged**